app/src/main/java/com/kms/cura/view/activity/BookAppointmentActivity.java

```diff
@@ -0,0 +1,227 @@
+        }
+        Dialog dateDialog = new DatePickerDialog(this, AlertDialog.THEME_HOLO_LIGHT, myDateListener, year, month - 1, day);
+        dateDialog.show();
+    }
+
+    private void chooseTime() {
+        if (spnApptLength.getSelectedItemPosition() != apptLength.size() - 1 &&
+                spnFacilities.getSelectedItemPosition() != doctorFacilityName.size() - 1 && selectDate) {
+            Intent toChooseTime = new Intent(this, ChooseTimeActivity.class);
+            Bundle bundle = createBundle();
+            toChooseTime.putExtras(bundle);
+            startActivityForResult(toChooseTime, REQUEST_cODE);
+        } else {
+            Toast.makeText(this, getResources().getString(R.string.BookAptError), Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    private DatePickerDialog.OnDateSetListener myDateListener = new DatePickerDialog.OnDateSetListener() {
+        @Override
+        public void onDateSet(DatePicker datePicker, int year, int month, int day) {
+            setSpecificDate(day, month + 1, year);
+        }
+    };
+
+    private void setCurrentDate() {
+        calendar = Calendar.getInstance();
+        day = calendar.get(Calendar.DAY_OF_MONTH);
+        month = calendar.get(Calendar.MONTH) + 1;
+        year = calendar.get(Calendar.YEAR);
+
+    }
+
+    private void setSpecificDate(int day, int month, int year) {
+        if(isDateSelectedInPast(day,month,year)){
+            ErrorController.showDialog(this,getResources().getString(R.string.BookAptDayError));
+            return;
+        }
+        this.day = day;
+        this.month = month;
+        this.year = year;
+        setDateString(day, month, year);
+    }
+
+    public void setDateString(int day, int month, int year) {
+        StringBuilder builder = new StringBuilder();
+        builder.append(" ");
+        if (month < 10)
+            builder.append("0");
+        builder.append(month + "/");
+        if (day < 10)
+            builder.append('0');
+        builder.append(day + "/");
+        builder.append(year);
+        txtDate.setText(builder.toString());
+        selectDate = true;
+    }
+
+    public Bundle createBundle() {
+        Bundle bundle = new Bundle();
+        bundle.putInt(DATE_SELECTED, day);
+        bundle.putInt(MONTH_SELECTED, month);
+        bundle.putInt(YEAR_SELECTED, year);
+        bundle.putInt(LENGTH_SELECTED, spnApptLength.getSelectedItemPosition() + 1);
+        return bundle;
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (requestCode == REQUEST_cODE) {
+            if (resultCode == RESULT_OK) {
+                Bundle time = data.getExtras();
+                txtTime.setText(time.getString(ChooseTimeActivity.START_TIME) + "-" + time.getString(ChooseTimeActivity.END_TIME));
+            }
+        }
+    }
+
+    private boolean isDateSelectedInPast(int day, int month, int year){
+        if(year >  this.year){
```

Comment:
you should just use the before or after method of Date class for date comparison without comparing it like this.